hw/arc/virt.c

```diff
@@ -163,26 +165,26 @@ static void virt_init(MachineState *machine)
 
     for (n = 0; n < VIRT_UART_NUMBER; n++) {
         serial_mm_init(system_io, VIRT_UART_OFFSET + VIRT_UART_SIZE * n, 2,
-                       cpu->env.irq[VIRT_UART_IRQ + n], 115200, serial_hd(n),
-                       DEVICE_NATIVE_ENDIAN);
+                       ARC_CPU(first_cpu)->env.irq[VIRT_UART_IRQ + n], 115200,
+                       serial_hd(n), DEVICE_NATIVE_ENDIAN);
     }
 
     for (n = 0; n < VIRT_VIRTIO_NUMBER; n++) {
         sysbus_create_simple("virtio-mmio",
                              VIRT_VIRTIO_BASE + VIRT_VIRTIO_SIZE * n,
-                             cpu->env.irq[VIRT_VIRTIO_IRQ + n]);
+                             ARC_CPU(first_cpu)->env.irq[VIRT_VIRTIO_IRQ + n]);
     }
 
-    create_pcie(cpu);
+    create_pcie(ARC_CPU(first_cpu));
 
-    arc_load_kernel(cpu, &boot_info);
+    arc_load_kernel(ARC_CPU(first_cpu), &boot_info);
 }
 
 static void virt_machine_init(MachineClass *mc)
 {
     mc->desc = "ARC Virtual Machine";
     mc->init = virt_init;
-    mc->max_cpus = 1;
+    mc->max_cpus = 64;
```

Comment:
ARCv2 processors may have only up-to 4 cores, ARCv3 up-to 12. Let's not make-up nonsense to not confuse our users :)